model/error/_meta/fields.yml

```diff
@@ -2,6 +2,767 @@
   title: APM Error
   description: Error-specific data for APM
   fields:
+    - name: data_stream.type
+      type: keyword
+      description: "Data stream type: logs, metrics, or traces."
+      example: traces
+
+    - name: data_stream.dataset
+      type: keyword
+      description: Data stream dataset name.
+      example: backend_service
+
+    - name: data_stream.namespace
+      type: keyword
+      description: User-defined data stream namespace.
+      example: production
```

Comment:
Not a big thing, but as soon as we add the `data_stream` fields here they will be in the template and show up at the docs. Do we really want this at this point? Why do we need the fields here already?

Reply:
This comes from merging @axw changes, so I'll summon him to answer this questions :)

(I personally think it is ok thou, as that is always the case for any new fields added - they show up in the master docs before the actual release)

Reply:
Right, we already have the fields in the templates. I got confused with having the fields in the dedicated [base-field.yml in the other PR](https://github.com/elastic/apm-server/pull/4432/files#diff-bb85cbda3ca728d641efb8225953c02e871930ebcd87a4ca7529fc2f309a1b99). 

Will take this to an offline discussion about why we introduced the fields.

Reply:
> I got confused with having the fields in the dedicated base-field.yml in the other PR.

Totally reasonable!

Reply:
I probably should not have added those fields as they're really only meaningful when running under Agent. We can remove them in a followup, as there will probably be some fallout in tests.